custom_components/sector/event.py

```diff
@@ -0,0 +1,140 @@
+"""Event platform for Sector Alarm integration."""
+
+import logging
+import asyncio
+import pytz
+
+from collections import defaultdict
+from homeassistant.components.event import EventEntity
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.event import async_call_later
+from datetime import datetime
+
+from .coordinator import SectorAlarmConfigEntry, SectorDataUpdateCoordinator
+from .entity import SectorAlarmBaseEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+class SectorAlarmEvent(SectorAlarmBaseEntity, EventEntity):
+    """Representation of a Sector Alarm log event."""
+
+    def __init__(self, coordinator: SectorDataUpdateCoordinator, device_serial: str, device_name: str, device_model: str):
+        """Initialize the log event entity for a specific device."""
+        super().__init__(coordinator, device_serial, {"name": device_name}, device_model)
+
+        self._attr_unique_id = f"{device_serial}_event"
+        self._attr_name = f"{device_name} Event Log"
+        self._attr_device_class = "timestamp"
+        self._events = []  # List to store recent log events for this device
+        self._initialized = False
+        _LOGGER.debug("Sector Entry: Created SectorAlarmEvent for device: %s with serial: %s", device_name, device_serial)
+
+    async def async_added_to_hass(self):
+        """Handle entity addition to Home Assistant."""
+        self._initialized = True  # Entity is now added to Home Assistant
+        _LOGGER.debug("Sector Entry: SectorAlarmEvent entity added to Home Assistant for: %s", self._attr_name)
+
+    @property
+    def state(self) -> str:
+        """Return the most recent event type as the state."""
+        return self._events[-1]["EventType"] if self._events else "No events"
+
+    @property
+    def extra_state_attributes(self) -> dict:
+        """Return additional attributes for the most recent log event."""
+        if not self._events:
+            return {}
+
+        recent_event = self._events[-1]
+
+        # Parse the ISO 8601 timestamp to a datetime object
+        time_str = recent_event.get("Time", "unknown")
+        try:
+            # Parse the string and localize it to UTC
+            timestamp = datetime.fromisoformat(time_str).astimezone(pytz.UTC)
```

Comment:
```suggestion
            timestamp = datetime.fromisoformat(time_str).astimezone(datetime.UTC)
```

Reply:
timezone.utc could be used. There isn't any datetime.UTC.

Reply:
Good. We should leave out third-party modules if we can 👍